src/node/node_state.h

```diff
@@ -1669,14 +1696,18 @@ namespace ccf
       // map the node id to a hostname and service and inform pbft
       network.nodes.set_local_hook(
         [this](kv::Version version, const Nodes::Write& w) {
-          for (auto& [node_id, ni] : w)
+          for (const auto& [node_id, opt_ni] : w)
           {
-            add_node(node_id, ni.value.nodehost, ni.value.nodeport);
+            if (!opt_ni.has_value())
+            {
+              throw std::logic_error("Unexpected: removal from nodes table");
```

Comment:
Not likely this'll happen, but if it does there's a small chance we may be happy to have the node id.